hal/inc/hal_dynalib_concurrent.h

```diff
@@ -61,6 +61,8 @@ DYNALIB_FN(24, hal_concurrent, os_queue_destroy, int(os_queue_t, void*))
 DYNALIB_FN(25, hal_concurrent, os_queue_put, int(os_queue_t, const void* item, system_tick_t, void*))
 DYNALIB_FN(26, hal_concurrent, os_queue_take, int(os_queue_t, void* item, system_tick_t, void*))
 DYNALIB_FN(27, hal_concurrent, os_thread_exit, os_result_t(os_thread_t))
+DYNALIB_FN(28, hal_concurrent, dct_lock, int(int, void*))
+DYNALIB_FN(29, hal_concurrent, dct_unlock, int(int, void*))
```

Comment:
This is mainly for electron, where DCT (or DCD) implementation is in `libplatform`, which gets linked into `system-part1`, causing linking error because FreeRTOS functions are not available there. Another solution for this could be implementing `os_mutex_recursive_create_ex` and using HAL functions for `RecursiveMutex` implementation.